addons/account/static/src/components/product_label_section_and_note_field/product_label_section_and_note_field.js

```diff
@@ -0,0 +1,152 @@
+/** @odoo-module **/
+
+import { _t } from "@web/core/l10n/translation";
+import { registry } from "@web/core/registry";
+import { useService } from "@web/core/utils/hooks";
+import { useState } from "@odoo/owl";
+
+import { Many2XAutocomplete } from "@web/views/fields/relational_utils";
+import { Many2OneField, many2OneField } from "@web/views/fields/many2one/many2one_field";
+import { X2ManyField, x2ManyField } from "@web/views/fields/x2many/x2many_field";
+import {
+    SectionAndNoteListRenderer,
+    sectionAndNoteFieldOne2Many,
+} from "@account/components/section_and_note_fields_backend/section_and_note_fields_backend";
+
+export class ProductLabelSectionAndNoteListRender extends SectionAndNoteListRenderer {
+    setup() {
+        super.setup();
+        this.titleField = "name";
+    }
+
+    getCellTitle(column, record) {
+        if (column.name == "product_id") {
+            return;
+        }
+        super.getCellTitle(column, record);
+    }
+
+    async toggleOptionalField(fieldName) {
+        super.toggleOptionalField(fieldName);
+        if (fieldName === "product_id") {
+            this.optionalActiveFields["name"] = !this.optionalActiveFields[fieldName];
+        } else if (fieldName === "name") {
+            this.optionalActiveFields["product_id"] = !this.optionalActiveFields[fieldName];
+        }
+        this.state.columns = this.getActiveColumns(this.props.list);
+        this.saveOptionalActiveFields(
+            this.allColumns.filter((col) => this.optionalActiveFields[col.name] && col.optional)
+        );
+    }
+
+    getIgnoredColumns() {
+        return ["name"];
+    }
+
+    getActiveColumns(list) {
+        const activeColumns = super.getActiveColumns(list);
+        this.titleField = activeColumns.some((col) => col.name === "product_id")
+            ? "product_id"
+            : "name";
+        return activeColumns;
+    }
+}
+
+export class ProductLabelSectionAndNoteOne2Many extends X2ManyField {
+    static components = {
+        ...X2ManyField.components,
+        ListRenderer: ProductLabelSectionAndNoteListRender,
+    };
+}
+
+export const productLabelSectionAndNoteOne2Many = {
+    ...x2ManyField,
+    component: ProductLabelSectionAndNoteOne2Many,
+    additionalClasses: sectionAndNoteFieldOne2Many.additionalClasses,
+};
+
+registry
+    .category("fields")
+    .add("product_label_section_and_note_field_o2m", productLabelSectionAndNoteOne2Many);
+
+export class ProductLabelSectionAndNoteFieldAutocomplete extends Many2XAutocomplete {
+    setup() {
+        super.setup();
+        this.orm = useService("orm");
+    }
+
+    async loadOptionsSource(request) {
+        const options = await super.loadOptionsSource(request);
+        if (request.length) {
+            options.unshift({
+                label: _t('Label "%s"', request),
+                classList: "o_m2o_dropdown_option",
+                action: () => {
+                    this.props.record.update({ name: request });
```

Comment:
Probably, you have to change for this:
```suggestion
                    this.props.update({ name: request });
```
You'll use the update from `Many2OneField` component.

Reply:
I decided to go with a binding of the method updateLabel from the class ProductLabelSectionAndNoteField so i can use it here.

I find it way cleaner as in the end you only have one update method (updateLabel) which does both updates from the autocomplete option and from writing and pressing enter in the input. I find it easy to read but I might be biased as i wrote it. :smile: Please tell me your opinion on this.